.github/workflows/continous-integration.yml

```
@@ -276,7 +276,107 @@ jobs:
         JOBS: 2
         PYTHONIOENCODING: "utf-8"
       run: |
-        make test
+        make test-non-training
+
+    - name: Send Coverage Report 📊
+      if: needs.changes.outputs.backend == 'true' && matrix.python-version == 3.6 && matrix.os != 'windows-latest'
+      env:
+        COVERALLS_REPO_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        COVERALLS_SERVICE_NAME: github
+      run: poetry run coveralls
+
+  test_training:
+    name: Run Tests That Train
+    runs-on: ${{ matrix.os }}
+    needs: [changes]
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest]
+        python-version: [3.6, 3.7, 3.8]
```

I think we would add here something like this

```suggestion
        python-version: [3.6, 3.7, 3.8]
        test: [test-non-training, test-training]
```

---

nice 👍